### PR TITLE
Support UDP transport

### PR DESCRIPTION
**Pull request by p0pr0ck5**

### CODE REVIEW

**Comment by p0pr0ck5** [comment]
ping? @calio @agentzh?

File: t/proto.t [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            local logger = require "resty.logger.socket"
+            if not logger.initted() then
+                local ok, err = logger.init{
+                    host = "127.0.0.1",
+                    port = 29999,
+                    sock_type = "udp",
+                    flush_limit = 1,
+                }
+
+				if err then ngx.log(ngx.ERR, err) end
```
Comment (calio) [inline t/proto.t]: Format here is weird.

File: t/proto.t [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            local logger = require "resty.logger.socket"
+            if not logger.initted() then
+                local ok, err = logger.init{
+                    host = "127.0.0.1",
+                    port = 29999,
+                    sock_type = "tcp",
+                    flush_limit = 1,
+                }
+
+				if err then ngx.log(ngx.ERR, err) end
```
Comment (calio) [inline t/proto.t]: Also here.

File: t/proto.t (hunk 1 above)
Comment (p0pr0ck5) [inline t/proto.t]: Whups, tabs not spaces. My bad.

**Comment by calio** [comment]
Merged with some modifications. Thank you for your contribution!

**Pull request closed without merging**